src/assets/scripts/client/aircraft/FlightManagementSystem/RouteModel.js

```diff
@@ -1094,6 +1210,15 @@ export default class RouteModel extends BaseModel {
         return amendedStarLeg;
     }
 
+    /**
+     * Return a STAR leg based on the provided leg, except with the new specified exit
+     *
+     * @for RouteModel
+     * @method _createAmendedStarLegUsingDifferentExitName
+     * @param exitFixName {string} name of STAR exit to use for the new STAR leg
+     * @param legIndex {number} index of leg in the #_legCollection
+     * @return {LegModel}
+     */
     _createAmendedStarLegUsingDifferentExitName(exitFixName, legIndex) {
         const divergentLegModel = this._legCollection[legIndex];
         const procedureIcao = divergentLegModel.getProcedureIcao();
```

Comment:
couple lines down, don't we have a helper somewhere that builds the procedure route string for us?  if not, we really should, cause this is used all over the place.

Reply:
Added (with test) and used here and elsewhere.